util/db/helmrepository.go

```diff
@@ -0,0 +1,74 @@
+package db
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/util/settings"
+)
+
+func getHelmRepoCredIndex(helmRepositories []settings.HelmRepoCredentials, repoURL string) int {
+	for i, cred := range helmRepositories {
+		if strings.EqualFold(cred.URL, repoURL) {
+			return i
+		}
+	}
+	return -1
+}
+
+func (db *db) getHelmRepo(repoURL string, helmRepositories []settings.HelmRepoCredentials) (*v1alpha1.Repository, error) {
+	index := getHelmRepoCredIndex(helmRepositories, repoURL)
+	if index < 0 {
+		return nil, status.Errorf(codes.NotFound, "repo '%s' not found", repoURL)
+	}
+	repoInfo := helmRepositories[index]
+
+	repo := &v1alpha1.Repository{
+		Repo:                  repoInfo.URL,
+		Type:                  "helm",
+		Name:                  repoInfo.Name,
+		InsecureIgnoreHostKey: false,
+		Insecure:              false,
+		EnableLFS:             false,
+	}
+	err := db.unmarshalFromSecretsStr(map[*string]*v1.SecretKeySelector{
+		&repo.Username:          repoInfo.UsernameSecret,
+		&repo.Password:          repoInfo.PasswordSecret,
+		&repo.TLSClientCertData: repoInfo.CertSecret,
+		&repo.TLSClientCertKey:  repoInfo.KeySecret,
```

Comment:
what about CA?

Reply:
Nice catch. It is impossible to support/migrate configured CA with the current design. 

The good thing that CA support does not really work in helm itself: https://github.com/helm/helm/issues/3384 . So unlikely anyone managed to configure custom CA in Argo CD and we don't have to worry about this case.